lib/notifications/EmailNotifier.ts

```diff
@@ -0,0 +1,336 @@
+import nodemailer from 'nodemailer';
+import type { Transporter } from 'nodemailer';
+import type { EmailConfig } from '../Config';
+import { satoshisToSatcomma } from '../DenominationConverter';
+import type Logger from '../Logger';
+import {
+  formatError,
+  getChainCurrency,
+  getLightningCurrency,
+  getSendingReceivingCurrency,
+  splitPairId,
+} from '../Utils';
+import { CurrencyType, OrderSide, SwapType } from '../consts/Enums';
+import type { AnySwap } from '../consts/Types';
+import type ChainSwapData from '../db/models/ChainSwapData';
+import ChannelCreation from '../db/models/ChannelCreation';
+import type ReverseSwap from '../db/models/ReverseSwap';
+import type Swap from '../db/models/Swap';
+import type { ChainSwapInfo } from '../db/repositories/ChainSwapRepository';
+import { msatToSat } from '../lightning/ChannelUtils';
+import type Service from '../service/Service';
+import type Sidecar from '../sidecar/Sidecar';
+import type WalletManager from '../wallet/WalletManager';
+
+class EmailNotifier {
+  private transporter?: Transporter;
+
+  // This is a hack to add trailing whitespace which is trimmed by default in some email clients
+  private static trailingWhitespace = '\n\n';
+
+  constructor(
+    private readonly config: EmailConfig | undefined,
+    private readonly logger: Logger,
+    private readonly service: Service,
+    private readonly walletManager: WalletManager,
+    private readonly sidecar: Sidecar,
+  ) {
+    if (!config) {
+      this.logger.debug('Email config is missing');
+      return;
+    }
```

Comment:
```suggestion
    private readonly config: EmailConfig,
    private readonly logger: Logger,
    private readonly service: Service,
    private readonly walletManager: WalletManager,
    private readonly sidecar: Sidecar,
  ) {
```

This code will never run because the `EmailNotifier` is not initialized when `config.enabled` is not true